src/util/file_watch.h

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (C) 2022 Red Hat Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef _FILE_WATCH_H
+#define _FILE_WATCH_H
+
+#include <stdbool.h>
+#include <talloc.h>
+#include <tevent.h>
+#ifdef HAVE_SYS_INOTIFY_H
+#include <sys/inotify.h>
+#endif
+
+
+typedef void (*fw_fn)(const char *filename, void *arg);
```

Comment:
This name isn't really telling.
Something around `event handler` would be probably better.

Reply:
Renamed to `fw_callback`.